src/providers/wfs/qgswfsdataitems.cpp

```diff
@@ -50,6 +55,76 @@ QgsWfsLayerItem::~QgsWfsLayerItem()
 {
 }
 
+QList<QMenu *> QgsWfsLayerItem::menus()
+{
+  QList<QMenu *> menus;
+
+  if ( mPath.startsWith( QLatin1String( "geonode:/" ) ) )
+  {
+    QAction *actionCopyStyle = new QAction( tr( "Copy Style" ), this );
+    connect( actionCopyStyle, &QAction::triggered, this, &QgsWfsLayerItem::copyStyle );
+
+    QMenu *menuStyleManager = new QMenu();
+    menuStyleManager->setTitle( tr( "Styles" ) );
+    menuStyleManager->addAction( actionCopyStyle );
+    menus << menuStyleManager;
+  }
+
+  return menus;
+}
+
+void QgsWfsLayerItem::copyStyle()
+{
+  QString layerUri = this->uri();
+
+  QgsGeoNodeConnection *connection = nullptr;
+  Q_FOREACH ( const QString &connName, QgsGeoNodeConnectionUtils::connectionList() )
+  {
+    connection = new QgsGeoNodeConnection( connName );
```

Comment:
this is leaking the previous value of connection. Can you make connection a unique_ptr to avoid any chance of leaks?